@SeanPM5 Thanks for your feedback, I totally agree !

I have some trouble with the pre-commit hook, which re-indents the source file, which is nice but here it causes problem with the rendering.
In [developer-tools-state.js (l.211)](https://github.com/home-assistant/home-assistant-polymer/pull/4259/files#diff-4dbcfc851b0f24f493ec2d49f3f12487R211), I added the attribute `id` to get its content. My code is:
```js
           <td id="attributes-[[entity.entity_id]]">[[attributeString(entity)]]</td>
```

But the auto formating hook changes it to this

```js
          <td id="attributes-[[entity.entity_id]]">
                [[attributeString(entity)]]
          </td>
```
Which should not cause any problem but when displaying the page, the cell with the attributes contains spaces before and after the content which is really ugly (I'll add a screenshot tonight), like if it is some non-breakable spaces.
I disabled the precommit hook so it does not changes the code, but it breaks the CI because of code styling:

```
/home/travis/build/home-assistant/home-assistant-polymer/src/panels/developer-tools/state/developer-tools-state.js

  211:56  error  Replace `[[attributeString(entity)]]` with `⏎················[[attributeString(entity)]]⏎··············`  prettier/prettier
```

I don't understand why the spaces are rendered, and how can I change the formatting of the code so that the 'prettifier' doesn't cause this ? I tried many ways of opening and closing tags, with id property but still no luck.

Can someone help or explain me why I have these behaviors ?

--- a/src/panels/developer-tools/state/developer-tools-state.js
+++ b/src/panels/developer-tools/state/developer-tools-state.js
@@ -160,7 +160,7 @@ class HaPanelDevState extends EventsMixin(LocalizeMixin(PolymerElement)) {
         <template is="dom-repeat" items="[[_entities]]" as="entity">
           <tr>
             <td>
-              <paper-menu-button close-on-activate="true">
+              <paper-menu-button close-on-activate>
                 <paper-icon-button
                   icon="hass:dots-vertical"
                   slot="dropdown-trigger"
@@ -172,42 +172,46 @@ class HaPanelDevState extends EventsMixin(LocalizeMixin(PolymerElement)) {
                   selected="{{selectedItem}}"
                 >
                   <paper-item on-click="entityMoreInfo">
-                    <paper-icon-button
-                      icon="hass:information-outline"
-                      alt="[[localize('ui.panel.developer-tools.tabs.states.more_info')]]"
-                      title="[[localize('ui.panel.developer-tools.tabs.states.more_info')]]"
-                    ></paper-icon-button>
-                    [[localize('ui.panel.developer-tools.tabs.states.more_info')]]
+                    <paper-icon-item>
+                      <ha-icon
+                        icon="hass:information-outline"
+                        slot="item-icon"
+                      ></ha-icon>
+                      [[localize('ui.panel.developer-tools.tabs.states.more_info')]]
+                    </paper-icon-item>
                   </paper-item>
 
                   <paper-item action="copyId" on-click="entityCopyValue">
-                    <paper-icon-button
-                      icon="hass:content-copy"
-                      alt="[[localize('ui.panel.developer-tools.tabs.states.copy_entity_id')]]"
-                      title="[[localize('ui.panel.developer-tools.tabs.states.copy_entity_id')]]"
-                    ></paper-icon-button>
-                    [[localize('ui.panel.developer-tools.tabs.states.copy_entity_id')]]
+                    <paper-icon-item>
+                      <ha-icon
+                        icon="hass:content-copy"
+                        slot="item-icon"
+                      ></ha-icon>
+                      [[localize('ui.panel.developer-tools.tabs.states.copy_entity_id')]]
+                    </paper-icon-item>
                   </paper-item>
 
                   <paper-item action="copyState" on-click="entityCopyValue">
-                    <paper-icon-button
-                      icon="hass:content-copy"
-                      alt="[[localize('ui.panel.developer-tools.tabs.states.copy_entity_state')]]"
-                      title="[[localize('ui.panel.developer-tools.tabs.states.copy_entity_state')]]"
-                    ></paper-icon-button>
-                    [[localize('ui.panel.developer-tools.tabs.states.copy_entity_state')]]
+                    <paper-icon-item>
+                      <ha-icon
+                        icon="hass:content-copy"
+                        slot="item-icon"
+                      ></ha-icon>
+                      [[localize('ui.panel.developer-tools.tabs.states.copy_entity_state')]]
+                    </paper-icon-item>
                   </paper-item>
 
                   <paper-item
                     action="copyAttributes"
                     on-click="entityCopyValue"
                   >
-                    <paper-icon-button
-                      icon="hass:content-copy"
-                      alt="[[localize('ui.panel.developer-tools.tabs.states.copy_entity_attribute')]]"
-                      title="[[localize('ui.panel.developer-tools.tabs.states.copy_entity_attribute')]]"
-                    ></paper-icon-button>
-                    [[localize('ui.panel.developer-tools.tabs.states.copy_entity_attribute')]]
+                    <paper-icon-item>
+                      <ha-icon
+                        icon="hass:content-copy"
+                        slot="item-icon"
+                      ></ha-icon>
+                      [[localize('ui.panel.developer-tools.tabs.states.copy_entity_attribute')]]
+                    </paper-icon-item>
                   </paper-item>
                 </paper-listbox>
               </paper-menu-button>
@@ -316,7 +320,7 @@ class HaPanelDevState extends EventsMixin(LocalizeMixin(PolymerElement)) {
     } else if (action === "copyState") {
       copy(ev.model.entity.state);
     } else if (action === "copyAttributes") {
-      copy(this.attributeString(ev.model.entity).replace(/\n/g, "<br />"));
+      copy(safeDump(ev.model.entity.attributes).replace(/\n/g, "<br />"));
     } else {
       return;
     }

--- a/src/panels/developer-tools/state/developer-tools-state.js
+++ b/src/panels/developer-tools/state/developer-tools-state.js
@@ -63,6 +63,10 @@ class HaPanelDevState extends EventsMixin(LocalizeMixin(PolymerElement)) {
           height: 24px;
           padding: 0;
         }
+        .entities paper-menu-button {
+          height: 24px;
+          padding: 0;
+        }
         .entities td:nth-child(3) {
           white-space: pre-wrap;
           word-break: break-word;
@@ -156,20 +160,26 @@ class HaPanelDevState extends EventsMixin(LocalizeMixin(PolymerElement)) {
         <template is="dom-repeat" items="[[_entities]]" as="entity">
           <tr>
             <td>
-              <paper-icon-button
-                on-click="entityMoreInfo"
-                icon="hass:information-outline"
-                alt="[[localize('ui.panel.developer-tools.tabs.states.more_info')]]"
-                title="[[localize('ui.panel.developer-tools.tabs.states.more_info')]]"
-              >
-              </paper-icon-button>
-              <paper-menu-button>
+              <paper-menu-button close-on-activate="true">
                 <paper-icon-button
-                  icon="hass:menu"
+                  icon="hass:dots-vertical"
                   slot="dropdown-trigger"
                   alt="menu"
                 ></paper-icon-button>
-                <paper-listbox slot="dropdown-content" role="listbox">
+                <paper-listbox
+                  slot="dropdown-content"
+                  role="listbox"
+                  selected="{{selectedItem}}"
+                >
+                  <paper-item on-click="entityMoreInfo">
+                    <paper-icon-button
+                      icon="hass:information-outline"
+                      alt="[[localize('ui.panel.developer-tools.tabs.states.more_info')]]"
+                      title="[[localize('ui.panel.developer-tools.tabs.states.more_info')]]"
+                    ></paper-icon-button>
+                    [[localize('ui.panel.developer-tools.tabs.states.more_info')]]
+                  </paper-item>
+
                   <paper-item action="copyId" on-click="entityCopyValue">
                     <paper-icon-button
                       icon="hass:content-copy"

--- a/src/panels/developer-tools/state/developer-tools-state.js
+++ b/src/panels/developer-tools/state/developer-tools-state.js
@@ -208,7 +208,7 @@ class HaPanelDevState extends EventsMixin(LocalizeMixin(PolymerElement)) {
               is="dom-if"
               if="[[computeShowAttributes(narrow, _showAttributes)]]"
             >
-              <td id="attributes-[[entity.entity_id]]">[[attributeString(entity)]]</td>
+              <td>[[attributeString(entity)]]</td>
             </template>
           </tr>
         </template>
@@ -306,10 +306,7 @@ class HaPanelDevState extends EventsMixin(LocalizeMixin(PolymerElement)) {
     } else if (action === "copyState") {
       copy(ev.model.entity.state);
     } else if (action === "copyAttributes") {
-      var td = this.shadowRoot.getElementById(
-        "attributes-" + ev.model.entity.entity_id
-      );
-      copy(td.textContent.replace(/\n/g, "<br />"));
+      copy(this.attributeString(ev.model.entity).replace(/\n/g, "<br />"));
     } else {
       return;
     }

--- a/src/translations/en.json
+++ b/src/translations/en.json
@@ -2043,10 +2043,10 @@
             "no_entities": "No entities",
             "more_info": "More Info",
             "alert_entity_field": "Entity is a mandatory field",
-            "copy_entity_id": "Copy entity ID",
-            "copy_entity_state": "Copy entity state",
-            "copy_entity_attribute": "Copy entity attributes",
-            "copied": "Copied!"
+            "copy_entity_id": "Copy ID",
+            "copy_entity_state": "Copy state",
+            "copy_entity_attribute": "Copy attributes",
+            "copied": "Copied to clipboard"
           },
           "templates": {
             "title": "Template",

